src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDiagnosticConverter.cs

```diff
@@ -72,6 +75,18 @@ internal static Range ConvertSpanToRange(SourceSpan sourceSpan, SourceText sourc
             };
 
             return range;
+
+            int NormalizeIndex(int index)
+            {
+                if (index >= sourceText.Length)
```

Comment:
Curious, what happens for empty files? Are there any real scenarios when an empty file will have a Razor diagnostic?

Reply:
Great question. Can't think of how that'd happen but it doesn't hurt to be extra certain. I'll do an update.